salt/orch/etcd-migrate.sls

```diff
@@ -0,0 +1,84 @@
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+update_grains:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_grains
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+       - salt: update_grains
+
+master-stop-services:
+  salt.state:
+    - tgt: 'roles:kube-master'
+    - tgt_type: grain
+    - sls:
+      - kube-apiserver.stop
```

Comment:
And what about `haproxy`? I'm not sure, but maybe it would be odd to keep `haproxy` running because users would users connecting to the API server would receive an error, right?

Reply:
Migration will be done only during maintenance window so users should be notified. HAproxy will recreate all connections to services on backend. To stop HAproxy we should stop docker services on all nodes, which will cause restart user application.